kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/UnreconciledDependencyException.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+/**
+ * Indicates that a dependency has not been reconciled, therefore we are operating on incomplete information.
+ * We cannot continue reconciliation as the state of that dependency is undetermined. This should be a transient
+ * state. The dependency should be reconciled, and it's updated status event should prompt the dependant reconciler
+ * to reconcile again, this time successfully.
+ */
+public class UnreconciledDependencyException extends RuntimeException {
```

Comment:
We've got two terms floating about, dependencies and referents. I think we should try to use one term. "Reference" and "referent" seem like the better choice since these things boil down to `*Ref` properties.

So I wonder if this should be `StaleReferentStatusException`?

Reply:
Agree, we already have referent (needs another PR to tidy up some other existing usages of dependency). Pushed up rename from:

dependency -> referent
unreconciled -> referentWithStaleStatus
reconciled -> referentWithFreshStatus